src/Agent/NewRelic/Agent/Core/DataTransport/HttpCollectorWire.cs

```diff
@@ -156,10 +156,14 @@ public string SendData(string method, ConnectionInfo connectionInfo, string seri
             }
         }
 
-        private static void AuditLog(Direction direction, Source source, string uri)
+        private void AuditLog(Direction direction, Source source, string uri)
         {
-            var message = string.Format(AuditLogFormat, direction, source, Strings.ObfuscateLicenseKeyInAuditLog(uri, LicenseKeyParameterName));
-            Logging.AuditLog.Log(message);
+            if (Logging.AuditLog.IsAuditLogEnabled)
+            {
+                var message = string.Format(AuditLogFormat, direction, source,
+                    Strings.ObfuscateLicenseKeyInAuditLog(uri, LicenseKeyParameterName));
+                Logging.AuditLog.Log(message);
```

Comment:
Should we add a test that verifies that we only attempt to generate the audit log when the AuditLogIsEnabled?

Reply:
Done.